### PR TITLE
Add `class-resolver`-based materialization of interface objects

### LINKED ISSUE #126: Instantiation pattern seems backwards
I'm not a big fan of this pattern:

```python
resource = OntologyResource(slug='go-nucleus.obo', directory='input', local=True)
oi = ProntoImplementation(resource)
```

It makes it seem like the implementation is based off of the resource itself, and not that the implementation is a "backend". I'd rather see code that looks like this:

```python
oi = get_interface(slug='go-nucleus.obo', directory='input', local=True, implementation="pronto")
# OR
oi = get_interface(slug='go-nucleus.obo', directory='input', local=True, implementation=ProntoImplementation)
```

In fact, we've written so many "backend"-like interfaces in [PyKEEN](https://github.com/pykeen/pykeen/) that we spun out the utilities we were using for that into a separate package [`class_resolver`](https://github.com/cthoyt/class-resolver) to support these kinds things (e.g., lookup of a class based on a string or class). Will send a PR to demonstrate

### PR DESCRIPTION
**Pull request by cthoyt**
Closes #126 

- [x] implements [`class-resolver`](https://github.com/cthoyt/class-resolver) lookup
- [x] Adds `class-resolver` as dependency (idk how poetry works and there's no contribution guide in this repo yet :p)

## Example usage

```python

from oaklib import OntologyResource

ontology_resource = OntologyResource(slug='tests/input/go-nucleus.db', local=True)

# Defaults to SQL implementation
ontology_interface = ontology_resource.materialize()

# Is smart and figures out which is the SQL one by name
# i.e., it knows to append "Implementation" at the end of the name
ontology_interface = ontology_resource.materialize("sql")

# Explicit
from oaklib.implementations import SqlImplementation
ontology_interface = ontology_resource.materialize(SqlImplementation)

# Forwards kwargs to SqlImplementation or whatever implementation you picked
from sqlalchemy import create_engine
engine = create_engine('postgresql://scott:tiger@localhost:5432/mydatabase')
ontology_interface = ontology_resource.materialize("sql", engine=engine)
```

### CODE REVIEW

**Comment by cmungall** [comment]
```bash
poetry add class-resolver
```

from inside the repo should work

**Comment by cthoyt** [comment]
I can't seem to get this to work on py310. py39 works alright.

**Comment by cmungall** [comment]
I am not sure we want to keep scipy as a core dependency

File: README.md [hunk 1]
```diff
@@ -39,17 +39,18 @@ All contributors are expected to uphold our [Code of Conduct](.github/CODE_OF_CO
...
+ontology_interface = ontology_resource.materialize("sql")
+# can also pass an implementation class explicitly instead of a string.
+
+for curie in ontology_interface.basic_search("cell"):
+    print(f'{curie} ! {ontology_interface.get_label_by_curie(curie)}')
+    for rel, fillers in ontology_interface.get_outgoing_relationship_map_by_curie(curie).items():
+        print(f'  RELATION: {rel} ! {ontology_interface.get_label_by_curie(rel)}')
         for filler in fillers:
-            print(f'     * {filler} ! {oi.label(filler)}')
+            print(f'     * {filler} ! {ontology_interface.get_label_by_curie(filler)}')
```
Comment (pkalita-lbl) [inline README.md:53]: I think `get_label_by_curie` has been deprecated in favor of the `label` method

**Review by pkalita-lbl — COMMENTED** [review]
Just one small comment about the `README` changes otherwise I think this looks good!

File: README.md (hunk 1 above)
Comment (cthoyt) [reply README.md]: Addressed! Let's merge this.

**Comment by cthoyt** [comment]
@pkalita-lbl please accept this so we can merge

**Review by pkalita-lbl — APPROVED** [review]

**Pull request merged**